libpromises/evalfunction.c

```diff
@@ -6990,6 +6990,10 @@ void ModuleProtocol(EvalContext *ctx, char *command, const char *line, int print
             StringSetAddSplit(tags, content, ',');
             StringSetAdd(tags, xstrdup("source=module"));
         }
+        else if (1 == sscanf(line + 1, "persistence=%ld", persistence))
```

Comment:
It looks like you may need to use a temp variable for ``persistence`` in ``scanf``, and then assign it after you know it's valid. The exotic platforms fail, and my best guess is that they start to parse ``persistence``, then fail, and in the meantime, they have added garbage to ``persistence``.

The output from AIX is:
```
R: /home/jenkins/workspace/testing-enterprise-pr_core/label/PACKAGES_ppc64_aix_5_3/core/tests/acceptance/./08_commands/01_modules/set-persistent-class.cf FAIL
R: dcs_passif: failing based on class "__passif"
R: dcs_passif_expected: bad: __passif_output_wanted was NOT defined
R: dcs_passif_output: bad: .*Module set persistent class "Cx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP" for 20 minutes.*Adding persistent class "Cx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP" to heap was NOT found in the output of ("/home/jenkins/workspace/testing-enterprise-pr_core/label/PACKAGES_ppc64_aix_5_3/core/tests/acceptance/workdir/__08_commands_01_modules_set-persistent-class_cf/bin/cf-agent" -Kvf /home/jenkins/workspace/testing-enterprise-pr_core/label/PACKAGES_ppc64_aix_5_3/core/tests/acceptance/./08_commands/01_modules/set-persistent-class.cf.sub; "/home/jenkins/workspace/testing-enterprise-pr_core/label/PACKAGES_ppc64_aix_5_3/core/tests/acceptance/workdir/__08_commands_01_modules_set-persistent-class_cf/bin/cf-promises" -v)| /bin/grep "x4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP"
R: dcs_passif_output: the output of command "("/home/jenkins/workspace/testing-enterprise-pr_core/label/PACKAGES_ppc64_aix_5_3/core/tests/acceptance/workdir/__08_commands_01_modules_set-persistent-class_cf/bin/cf-agent" -Kvf /home/jenkins/workspace/testing-enterprise-pr_core/label/PACKAGES_ppc64_aix_5_3/core/tests/acceptance/./08_commands/01_modules/set-persistent-class.cf.sub; "/home/jenkins/workspace/testing-enterprise-pr_core/label/PACKAGES_ppc64_aix_5_3/core/tests/acceptance/workdir/__08_commands_01_modules_set-persistent-class_cf/bin/cf-promises" -v)| /bin/grep "x4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP"" was: +Cx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP""
 verbose: Activating classes from module protocol: "Cx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP"
 verbose: Module set persistent class "Cx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP" for 86437032304 minutes
 verbose: Updating persistent class "Cx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP"
+Cx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP"" -- succeeded
+Cx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP""
+Bx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP""
 verbose: Activating classes from module protocol: "Bx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP"
+Bx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP"" -- succeeded
+Bx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP""
 verbose: P:    Container path : "/default/main/commands/"/bin/echo "+Ax4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP""[0]"
    info: Executing "no timeout" ... "/bin/echo "+Ax4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP""
 verbose: Activating classes from module protocol: "Ax4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP"
 verbose: Finished command related to promiser "/bin/echo "+Ax4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP"" -- succeeded
    info: Completed execution of "/bin/echo "+Ax4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP""
 verbose: Persistent class "Cx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP" for 19 more minutes
 verbose: Adding persistent class "Cx4eXfWrHT0zsSbrAXh5jFnRnDLKqNsP" to heap
```

As you see, number of minutes is way off.